lib/std/json.zig

```diff
@@ -1201,6 +1201,36 @@ pub const Value = union(enum) {
         w.one_indent = one_indent;
         try w.emitJson(self);
     }
+
+    /// json.Value equality comparison: for `.Array`s and `.Object`s, equal
+    /// sizes are prerequisite before further probing into their contents.
+    pub fn eql(self: Value, other: Value) bool {
+        if (std.meta.activeTag(self) == std.meta.activeTag(other)) switch (self) {
+            .Null => return true,
+            .Bool => |self_bool| return self_bool == other.Bool,
+            .Integer => |self_int| return self_int == other.Integer,
+            .Float => |self_float| return self_float == other.Float, // TODO: reconsider if/when std.math gets a dedicated float eql
+            .String => |self_string| return mem.eql(u8, self_string, other.String),
+
+            .Array => |self_array| if (self_array.len == other.Array.len) {
```

Comment:
This falls through when array length doesn't match.
Would be better to just do `if (self_array.len != other.Array.len) return false;`

Reply:
Oh curious, hadn't considered the fallthrough possibility since in my other langs used, the general semantics is "once a prong matches, others won't be hit". Done. in https://github.com/ziglang/zig/pull/4365/commits/cf7ce2f030b0db9a6cb7557d20277ea7490f0772

Reply:
It also applies to the `Object` path.

Reply:
What am I missing here: `.Object` it's the last prong, so it "falls through" the final return. Or can the prongs become reordered behind the curtains?

Reply:
Oh, in zig switch statements don't fall through like in C. I was just suggesting to move the `return` false inside the Object case.

Reply:
OK can do, but the final one is still needed because of the activeTag cmp

Reply:
Could change the first line to `if (std.meta.activeTag(self) != std.meta.activeTag(other)) return false;` and then put an `unreachable` at the end of the switch

Reply:
The latter would be utterly superfluous then. But I'll indulge you. It's for std after all.